src/recipe/Baked.Recipe.Service/Runtime/FileProviderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Text;
+
+namespace Microsoft.Extensions.FileProviders;
+
+public static class FileProviderExtensions
+{
+    public static string? ReadAsString(this IFileProvider provider, string subPath)
+    {
+        var fileInfo = provider.GetFileInfo(subPath);
+        if (!fileInfo.Exists) { return null; }
+
+        using var stream = fileInfo.CreateReadStream();
+        using var streamReader = new StreamReader(stream, Encoding.UTF8);
```

Comment:
these lines are duplicated. add another extension method (don't make it public) `provider.CreateStreamReader`

Reply:
you've only extracted file exists secion. i suppose you assume stream isn't closing when the reader closes. afaik, reader closes underlying stream, so you don't need to place the stream in a using statement, meaning you can extract it to the new method as well.

did you check your assumption?

Reply:
Checked the documentation and code, `StreamReader` is closing the underlying stream unless `leaveOpen` parameter is set to `true`, so moved to the new method